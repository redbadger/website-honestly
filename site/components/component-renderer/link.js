// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
