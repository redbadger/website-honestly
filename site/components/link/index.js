// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
