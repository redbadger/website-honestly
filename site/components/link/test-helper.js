// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
