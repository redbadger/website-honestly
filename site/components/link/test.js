// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
