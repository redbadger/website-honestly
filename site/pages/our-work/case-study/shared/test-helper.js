// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
