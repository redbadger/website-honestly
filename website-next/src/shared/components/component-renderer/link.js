import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
