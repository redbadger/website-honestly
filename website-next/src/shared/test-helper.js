import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
