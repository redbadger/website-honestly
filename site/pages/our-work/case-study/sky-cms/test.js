import React from 'react';
import { render } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import SkyCmsCaseStudy from './index';

jest.mock(
  '../shared/what-to-read-next',
  () => require.requireActual('../shared/test-helper').default,
);

describe('site/pages/our-work/case-study/sky-cms/', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <SkyCmsCaseStudy />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
