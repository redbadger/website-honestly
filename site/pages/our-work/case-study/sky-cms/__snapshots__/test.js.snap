// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/sky-cms/ renders correctly 1`] = `
<div
  class="caseStudy"
>
  <div
    class="header"
  >
    <div
      class="header__container"
    >
      <div
        class="header__imageContainer"
      >
        <img
          alt="Sky CMS"
          class="header__image"
          src="header.png"
        />
      </div>
    </div>
  </div>
  <div
    class="content"
  >
    <div
      class="content__mainTitleWrapper"
    >
      <h1
        class="content__mainTitle"
      >
        Bespoke and made from scratch: a new CMS just for Sky
      </h1>
    </div>
    <div
      class="overview"
    >
      <div
        class="quotation quote"
      >
        <div
          class="quotation__text"
        >
          <blockquote
            class="quotation__quote"
          >
            The new site and CMS have given us the flexibility we needed to effectively update and manage the Help site. It allows us to take an agile approach to the way we work and respond to consumer requirements faster and more efficiently.
          </blockquote>
        </div>
        <div
          class="quotation__author"
        >
          <div
            class="avatar__container"
            style="width:70px;height:70px;min-width:70px;min-height:70px"
          >
            <img
              alt="David Crawford"
              src="quote-author.jpg"
            />
          </div>
          <div
            class="quotation__author__wrapper"
          >
            <div
              class="quotation__author__name"
            >
              — David Crawford
            </div>
            <div
              class="quotation__author__title"
            >
              Head of Software Engineering, Sky
            </div>
          </div>
        </div>
      </div>
      <div
        class="Card listBoxContainer"
      >
        <div
          class="listBox"
        >
          <h3
            class="listBox__heading"
          >
            Results
          </h3>
          <ul>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                New intuitive CMS creating more efficient system, freeing up time to focus on the content 
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                Tripled the number of issues resolved through online self-help journey
              </div>
              <div />
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="content__wrapper"
    >
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Let’s make things better.
        </span>
        Enabling Sky to manage their content with an amazing customer experience
      </h2>
      <p
        class="content__paragraph"
      >
        A CMS is responsible for three things: storing content, editing content, and publishing content. These are three distinct functions with their own challenges and should be built as such. If you get this part right and built in a reusable way, specifics can be built on top of them.
      </p>
      <p
        class="content__paragraph"
      >
        As part of its commitment to customer service excellence, Sky wanted to optimize their online self-help service and make it easier for their customers to find solutions to their problems. In a modern world where customer behavior is driving constantly changing requirements and an expectation of brilliant customer experience, it is vital that both a CMS and the website it drives are adaptable to these demands.
      </p>
      <p
        class="content__paragraph"
      >
        Taking off the shelf products such as Drupal and trying to bend them to your requirements is simply not good enough anymore. They are far too rigid to cope with ever changing demands and inhibit customer experience.
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Do the right thing. Do the thing right.
        </span>
        Bespoke customisation tools for a built-from-scratch CMS
      </h2>
      <p
        class="content__paragraph"
      >
        Red Badger worked with Sky to conduct an objective build vs buy exercise. Analysis of the available tools revealed that the customisation required to make an off-the-shelf tool fit for purpose meant that greater efficiencies could be achieved by implementing a bespoke solution tailored to their business requirements, leveraging an open-source foundation.
      </p>
      <div
        class="contactBox"
      >
        <h2
          class="contactBox__heading"
        >
          Project in mind?
        </h2>
        <button
          class="contactBox__button"
          type="button"
        >
          Tell us more
        </button>
      </div>
      <p
        class="content__paragraph"
      >
        The decision was made to build a new CMS from scratch with three main components to deal with the individual concerns; storing, editing and publishing content. The result is a modern CMS that fits Sky’s requirements exactly - one that would be the foundation of a whole new Help site that could support both customers and the internal content team far more effectively. As demand changes, new features can also be rolled out faster and cheaper due to reduced interdependence and complexity.
      </p>
      <h3
        class="content__secondaryTitle"
      >
        Innovative features of the CMS include:
      </h3>
      <div
        class="content__paragraph"
      >
        <ul
          class="content__ordered__list"
        >
          <li
            class="content__ordered__list__element"
          >
            An editor to build content out of blocks - pluggable components for different types of content (text, image, video, tabs)
          </li>
          <li
            class="content__ordered__list__element"
          >
            A very rich editing experience tailored to Sky’s needs
          </li>
          <li
            class="content__ordered__list__element"
          >
            Full audit mechanism which makes it easy to revert to previous versions
          </li>
          <li
            class="content__ordered__list__element"
          >
            Continuous Deployment pipeline to publish new features into live in seconds
          </li>
          <li
            class="content__ordered__list__element"
          >
            Responsive Web Design support with preview on desktop, mobile and tablet
          </li>
          <li
            class="content__ordered__list__element"
          >
            ElasticSearch integration for fast and flexible search
          </li>
          <li
            class="content__ordered__list__element"
          >
            Image hub to allow content editors to add and store images to the site, reusable for different purposes
          </li>
        </ul>
      </div>
      <p
        class="content__paragraph"
      >
        <img
          alt="cms illustration"
          class="cmsIllustration__image"
          src="cms-illustration.jpg"
        />
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Creating lasting change.
        </span>
        A system so good, it’s being adopted across the business
      </h2>
      <p
        class="content__paragraph"
      >
        The new CMS enables editors to provide users with the content they need, fast and efficiently. It has created a greatly enhanced customer experience on the Sky Help site and reduced the time (and skills) required to update and amend content. It is also future proofed, being incredibly easy to scale as requirements change.
      </p>
      <p
        class="content__paragraph"
      >
        User behavior is closely monitored so that any issues are made apparent and dealt with straight away. In one example, Sky were able to diagnose an issue as it arose, publish content to solve it, and drop the number of customers pushing the ‘need more help’ button by 50%.
      </p>
      <p
        class="content__paragraph"
      >
        The components on which the CMS is built have proved to be so effectual that they are now used to support other Sky sites and were made available to everyone as an open source software.
      </p>
    </div>
  </div>
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work"
    >
      See more of our work
    </a>
  </div>
  <section
    class="contactUsContainer"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>
</div>
`;
